odl/set/domain.py

```diff
@@ -154,9 +154,37 @@ def max(self):
         """The maximum value in this interval product"""
         return self.end
 
-    def element(self):
-        """An arbitrary element, the midpoint."""
-        return self.midpoint
+    def element(self, inp=None):
+        """Create element in this set.
+
+        Parameters
+        ----------
+        inp : `float` or array-like, optional
+            Point to be cast to an element in self
+
+        Returns
+        -------
+        element
+            Returns ``inp`` if given, else ``self.midpoint``
+
+        Raises
+        ------
+        TypeError
+            If ``inp`` is not a valid element.
+
+        Examples
+        --------
+        >>> interv = IntervalProd(0, 1)
+        >>> interv.element(0.5)
+        0.5
+        """
+        if inp is None:
+            return self.midpoint
+        elif inp in self:
+            return inp
+        else:
+            raise TypeError('inp {!r} not a valid element in {!r}.'
+                            ''.format(inp, self))
```

Comment:
Consistency, the definition of `element`says it should take an `inp`:

https://github.com/odlgroup/odl/blob/eec730cd0128a77a4315bf2303da755878bc346d/odl/set/sets.py#L126-L131